cli/commands/rewards/distribute.ts

```diff
@@ -0,0 +1,66 @@
+import { flags } from '@oclif/command'
+import fs from 'fs'
+import { BaseCommand } from '../../base'
+import MerkleDistributor from '../../MerkleDistributor.json'
+
+export default class DeployMerkleDistributor extends BaseCommand {
+  static description = 'Deploy Merkle Distributor Contract'
+
+  static flags = {
+    merkleTree: flags.string({ required: true, description: 'JSON file with rewards' }),
+    address: flags.string({ required: true, description: 'Address of merkle distributor contract' }),
+    env: flags.string({ required: false, description: '[default: local] Blockchain environment with which to interact' }),
+    from: flags.string({ required: true, description: 'Deployer address' }),  
+    privateKey: flags.string({ required: false, description: 'Use a private key to sign local transactions with' }), 
+  }
+
+  async run() {
+    const res = this.parse(DeployMerkleDistributor)
+    const merkleTree = JSON.parse(fs.readFileSync(res.flags.merkleTree, { encoding: 'utf8' }))
+    const from: string = res.flags.from.toLowerCase()
+    const gasPrice = await this.getGasPrice(this.kit)
+    const abi = MerkleDistributor.abi 
+    const distAddress = res.flags.address
+    const merkleDistributor = new this.kit.web3.eth.Contract(abi, distAddress)
+    const distributorMerkleRoot = await merkleDistributor.methods.merkleRoot().call()
+    if (merkleTree.merkleRoot != distributorMerkleRoot) {
+      this.error(`merkle roobt: ${merkleTree.merkleRoot} does not match contract root: ${distributorMerkleRoot}`)
+    }
+
+    const distributionFile = `distribution-${distAddress}.json`
+    if (!fs.existsSync(distributionFile) || fs.readFileSync(distributionFile).length == 0) {
+      fs.writeFileSync(distributionFile, '{}')
+    }
+    
+    for (let account of Object.keys(merkleTree.claims)) {
+      let claim = merkleTree.claims[account]
+      merkleDistributor.methods.isClaimed(claim.index).call((err: Error, claimed: boolean) =>{
+        if (err) this.error(err)
+        if (!claimed) {
+          merkleDistributor.methods.claim(
+            claim.index,
+            account,
+            claim.amount,
+            claim.proof
+          ).send(
+            { from, gas: 1500000, gasPrice }
+          ).on('receipt', (receipt: any) => {
```

Comment:
Is it not a problem here that you are not awaiting these calls? I.e. you will run quite a few of them in parallel and shouldn't the script exit prior to the code actually finishing?

Reply:
1) I want to run these in parallel or else I'd need to wait for a block to mine in between, leading this to take as long as 2 hours.
2) I don't exit the script when finishing, bc I want to collect the transactionHashes, so I do that via callback.

It usually takes no longer than 2 minutes to complete all transactions. Let me know if this answers the question sufficiently!